### PR TITLE
DOC-894: Update look and feel of Node base file page

### PR DESCRIPTION
**Pull request by freakwriter**
Includes these updates:

- Transitioned single file to directory format with top matter becoming index and each h2 becoming its own page.
- Add more cross-references between pages.
- Updated nav.
- Added cross-references between sections.

### CODE REVIEW

**Comment by freakwriter** [comment]
@StarfallProjects as discussed, I've created a subdirectory and split the h2s out into individual pages. Let me know what you think.

I briefly toyed around with manually adding HTML to achieve something similar to the data transformation functions, but quickly realized that was likely going to be more work than using the .yaml file and plugin, so I backed out those changes. I went back and forth on keeping the programmatic-style execute method as its own page, since the programmatic-style parameters is a pretty small page to begin with. Let me know if you think it would be better to collapse into a single page.

I wasn't 100% certain what metadata to use at the top of each page; I used a couple other pages for reference and did my best, but please review that especially.

I don't think a redirect for the removed single-page is necessary, since the URL ends up being the URL for the subdirectory itself, which feels correct, but let me know if I made any faulty assumptions there.

File: docs/integrations/creating-nodes/build/reference/node-base-files/index.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+These documents give short code snippets to help understand the code structure and concepts. For full walk-throughs of building a node, including real-world code examples, refer to [Build a declarative-style node](/integrations/creating-nodes/build/declarative-style-node/) or [Build a programmatic-style node](/integrations/creating-nodes/build/programmatic-style-node/).
+
+You can also explore the [n8n-nodes-starter](https://github.com/n8n-io/n8n-nodes-starter){:target=_blank .external-link} and n8n's own [nodes](https://github.com/n8n-io/n8n/tree/master/packages/nodes-base/nodes){:target=_blank .external-link} for a wider range of examples. The starter contains basic examples that you can build on. The n8n [Mattermost node](https://github.com/n8n-io/n8n/tree/master/packages/nodes-base/nodes/Mattermost) is a good example of a more complex programmatic-style node, including versioning.
+
+For all nodes, refer to the:
+
+* [Structure of the node base file](/integrations/creating-nodes/build/reference/node-base-files/structure/).
```
Comment (StarfallProjects) [inline docs/integrations/creating-nodes/build/reference/node-base-files/index.md:15]:
```suggestion
* [Structure of the node base file](/integrations/creating-nodes/build/reference/node-base-files/structure/)
```

File: docs/integrations/creating-nodes/build/reference/node-base-files/index.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+* [Standard parameters](/integrations/creating-nodes/build/reference/node-base-files/standard-parameters/).
```
Comment (StarfallProjects) [inline docs/integrations/creating-nodes/build/reference/node-base-files/index.md:16]:
```suggestion
* [Standard parameters](/integrations/creating-nodes/build/reference/node-base-files/standard-parameters/)
```

File: docs/integrations/creating-nodes/build/reference/node-base-files/index.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+For declarative-style nodes, refer to the:
+
+* [Declarative-style parameters](/integrations/creating-nodes/build/reference/node-base-files/declarative-style-parameters/).
```
Comment (StarfallProjects) [inline docs/integrations/creating-nodes/build/reference/node-base-files/index.md:20]:
```suggestion
* [Declarative-style parameters](/integrations/creating-nodes/build/reference/node-base-files/declarative-style-parameters/)
```

File: docs/integrations/creating-nodes/build/reference/node-base-files/index.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+For programmatic-style nodes, refer to the:
+
+* [Programmatic-style parameters](/integrations/creating-nodes/build/reference/node-base-files/programmatic-style-parameters/).
```
Comment (StarfallProjects) [inline docs/integrations/creating-nodes/build/reference/node-base-files/index.md:24]:
```suggestion
* [Programmatic-style parameters](/integrations/creating-nodes/build/reference/node-base-files/programmatic-style-parameters/)
```

File: docs/integrations/creating-nodes/build/reference/node-base-files/index.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+* [Programmatic-style execute() method](/integrations/creating-nodes/build/reference/node-base-files/programmatic-style-execute-method/).
```
Comment (StarfallProjects) [inline docs/integrations/creating-nodes/build/reference/node-base-files/index.md:25]:
```suggestion
* [Programmatic-style execute() method](/integrations/creating-nodes/build/reference/node-base-files/programmatic-style-execute-method/)
```

**Review by StarfallProjects — APPROVED** [review]

**Pull request merged**